File: doc/_i18n/jp/tutorials/introduction/multi-robot-controller/logic_setup.cpp
// ヘッダー内
enum DoorPhase
{
      APPROACH = 0,
      HANDLE,
      OPEN
};
// コントローラーのプライベートプロパティ
DoorPhase phase = APPROACH;
// コントローラーの新しいメソッド
void switch_phase()
{
  if(phase == APPROACH && 0 /** この条件は後で記述する */)
  {
    /** HANDLEフェーズをセットアップする */
    phase = HANDLE;
  }
  else if(phase == HANDLE && 0 /** この条件は後で記述する */)
  {
    /** OPENフェーズをセットアップする */
    phase = OPEN;
  }
}
// run関数内でこれを呼び出す
bool MyFirstController::run()
{
  switch_phase();
  return mc_control::MCController::run();
}

File: doc/_i18n/jp/tutorials/introduction/multi-robot-controller/logic_setup.py
# 定数を宣言する
APPROACH = 0
HANDLE = 1
OPEN = 2
# コンストラクター内
self.phase = APPROACH
# コントローラーの新しいメソッド
def switch_phase(self):
  if self.phase == APPROACH and False: # この条件は後で記述する
    # HANDLEフェーズをセットアップする
    self.phase = HANDLE
  elif self.phase == HANDLE and False: # この条件は後で記述する
    # OPENフェーズをセットアップする
    self.phase = OPEN
# runコールバック関数内でこれを呼び出す
def run_callback(self):
    self.switch_phase()
    return True
